### PR TITLE
chore: use `import type` where possible

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [ ] I branched from develop
- [ ] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

### Resolves

N/A

### Proposed Changes

Changes `import`s which were `goog.requireType`s in `goog.module`-land to `import type`s so that our dependency graph is simplified.

#### Behavior Before Change

No change in behavior.

#### Behavior After Change

No change in behavior.

### Reason for Changes

Simplify dependency graph.

### Additional Information

N/A

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -23,5 +23,9 @@
     "moduleResolution": "node",
     "target": "ES2020",
     "strict": true,
+    "outDir": "build",
```
Comment (cpcallen) [inline tsconfig.json:26]: Why specify this here?  Is the parameter supplied by `buildJavaScript` in `build_tasks.js` insufficient?

File: tsconfig.json [hunk 2]
```diff
@@ -23,5 +23,9 @@
...
+
+    // This does not work understand enums so we cannot leave it enabled.
+    // "importsNotUsedAsValues": "error"
```
Comment (cpcallen) [inline tsconfig.json:29]: Great to document, but documentation is not great: "does not work understand"??

Also: is it just getting upset about how we reexport enums from `namespaces`, or is this feature somehow fundamentally broken in some way that the TypeScript maintainers should be appraised of?

**Review by cpcallen — APPROVED** [review]

File: tsconfig.json (hunk 1 above)
Comment (BeksOmega) [reply tsconfig.json]: Removed, was testing with `tsc` instead of `npm run build:js`. But not necessary otherwise.

File: tsconfig.json (hunk 2 above)
Comment (BeksOmega) [reply tsconfig.json]: Fixed, it is not a problem with `namespaces`. I believe Christopher is digging into the actual issue, and filing a bug.

Comment (BeksOmega) [reply tsconfig.json]: Added a comment linking to https://github.com/microsoft/TypeScript/issues/49974

**Pull request merged**